beacon_node/lighthouse_network/src/peer_manager/peerdb/peer_info.rs

```diff
@@ -100,14 +102,16 @@ impl<E: EthSpec> PeerInfo<E> {
                         .syncnets()
                         .map_or(false, |s| s.get(**id as usize).unwrap_or(false))
                 }
-                Subnet::DataColumn(_) => {
-                    // TODO(das): Pending spec PR https://github.com/ethereum/consensus-specs/pull/3821
-                    // We should use MetaDataV3 for peer selection rather than
-                    // looking at subscribed peers (current behavior). Until MetaDataV3 is
-                    // implemented, this is the perhaps the only viable option on the current devnet
-                    // as the peer count is low and it's important to identify supernodes to get a
-                    // good distribution of peers across subnets.
-                    return true;
+                Subnet::DataColumn(column) => {
+                    let Ok(custody_subnet_count) = meta_data.custody_subnet_count() else {
+                        return false;
+                    };
+                    // Supernode will be subscribed to all subnets
+                    if *custody_subnet_count == spec.data_column_sidecar_subnet_count {
+                        true
```

Comment:
Why is it necessary to handle the super node case differently? Just have a hashset with all subnets. No need to pass the chainspec then